tests/test-extensions/reconnection-ext/src/main/java/org/mule/extension/test/extension/reconnection/FallibleReconnectableSource.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.test.extension.reconnection;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.connection.ConnectionProvider;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.scheduler.Scheduler;
+import org.mule.runtime.api.scheduler.SchedulerService;
+import org.mule.runtime.api.util.concurrent.Latch;
+import org.mule.runtime.extension.api.annotation.param.Connection;
+import org.mule.runtime.extension.api.runtime.operation.Result;
+import org.mule.runtime.extension.api.runtime.source.Source;
+import org.mule.runtime.extension.api.runtime.source.SourceCallback;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.inject.Inject;
+
+public class FallibleReconnectableSource extends Source<Void, Void> {
+
+  public static volatile boolean fail = false;
+  public static volatile boolean simultaneouslyStartedSources = false;
+
+  @Connection
+  ConnectionProvider<ReconnectableConnection> connectionProvider;
+
+  @Inject
+  SchedulerService schedulerService;
+
+  private Scheduler scheduler;
+  private ScheduledFuture<?> scheduleWithFixedDelay;
+
+  private static final AtomicInteger countStartedSources = new AtomicInteger(0);
+  private static final Latch latch = new Latch();
+
+  @Override
+  public void onStart(SourceCallback<Void, Void> sourceCallback) throws MuleException {
+    if (countStartedSources.addAndGet(1) > 1) {
```

Comment:
Shouldn't this verify that there are two countedStartedSources simultaneously in the onStart instead of substracting in the onStop. What you need to verify is that the onStart is not executed again until the firs one ended.